test/e2e/foursquare.e2e.test.js

```diff
@@ -0,0 +1,116 @@
+const { test, expect } = require('@playwright/test');
+
+test.describe('Foursquare Places API Integration', () => {
+  let sharedPage;
+
+  test.beforeAll(async ({ browser }) => {
+    sharedPage = await browser.newPage();
+    await sharedPage.goto('/api/foursquare');
+    await sharedPage.waitForLoadState('networkidle');
+  });
+
+  test.afterAll(async () => {
+    if (sharedPage) await sharedPage.close();
+  });
+
+  test('should load page and show primary UI elements', async () => {
+    // Document title and main header
+    await expect(sharedPage).toHaveTitle(/Foursquare/i);
+    await expect(sharedPage.locator('h2')).toContainText('Foursquare API');
+
+    // Top action buttons
+    const devInfo = sharedPage.locator('a.btn', { hasText: 'Developer Info' });
+    const apiDocs = sharedPage.locator('a.btn', { hasText: 'API Docs' });
+    await expect(devInfo).toBeVisible();
+    await expect(devInfo).toHaveAttribute('href', 'https://foursquare.com/developer');
+    await expect(apiDocs).toBeVisible();
+    await expect(apiDocs).toHaveAttribute('href', 'https://docs.foursquare.com/');
+
+    // Ensure no error alert is displayed when a valid key is configured
+    await expect(sharedPage.locator('.alert.alert-danger')).toHaveCount(0);
+  });
+
+  test('should render Trending Venues table with data', async () => {
+    // Section header and hint text
+    await expect(sharedPage.locator('h3.text-primary', { hasText: 'Trending Venues' })).toBeVisible();
+    await expect(sharedPage.locator('p', { hasText: 'Near longitude' })).toBeVisible();
+
+    // Table basics
+    const table = sharedPage.locator('table.table.table-striped.table-bordered');
+    await expect(table).toBeVisible();
+
+    const headers = table.locator('thead th');
+    await expect(headers).toHaveCount(5);
+    await expect(headers.nth(1)).toContainText('Name');
+    await expect(headers.nth(2)).toContainText('Category');
+    await expect(headers.nth(3)).toContainText('Address');
+    await expect(headers.nth(4)).toContainText('Distance');
+
+    // At least one result row
```

Comment:
The hard coded location is a major/busy tourist spot Downtown Seattle.  That location should always max out on results.  So if we are not hitting the limit of 10, there is probably something wrong.

`https://places-api.foursquare...&limit=10`